zookeeper-recipes/zookeeper-recipes-queue/src/test/java/org/apache/zookeeper/recipes/queue/DistributedQueueTest.java

```diff
@@ -27,243 +28,260 @@
     }
 
-    public void createNremoveMelementTest(String dir, int n, int m) throws Exception {
-        String testString = "Hello World";
-        final int numClients = 2;
-        ZooKeeper[] clients = new ZooKeeper[numClients];
-        DistributedQueue[] queueHandles = new DistributedQueue[numClients];
-        for (int i = 0; i < clients.length; i++) {
-            clients[i] = createClient();
-            queueHandles[i] = new DistributedQueue(clients[i], dir, null);
-        }
+    final byte[] takeResult[] = new byte[1][];
+    Thread takeThread = new Thread() {
+      public void run() {
+        try {
+          takeResult[0] = queueHandles[0].take();
+        } catch (KeeperException e) {
 
-        for (int i = 0; i < n; i++) {
-            String offerString = testString + i;
-            queueHandles[0].offer(offerString.getBytes());
-        }
+        } catch (InterruptedException e) {
 
-        for (int i = 0; i < m; i++) {
-            queueHandles[1].remove();
         }
-        Assert.assertEquals(new String(queueHandles[1].element()), testString + m);
-    }
+      }
+    };
+    takeThread.start();
 
-    @Test
-    public void testElement1() throws Exception {
-        createNremoveMelementTest("/testElement1", 1, 0);
-    }
+    Thread.sleep(1000);
+    Thread offerThread = new Thread() {
+      public void run() {
+        try {
+          queueHandles[0].offer(testString.getBytes(UTF_8));
+        } catch (KeeperException e) {
 
-    @Test
-    public void testElement2() throws Exception {
-        createNremoveMelementTest("/testElement2", 10, 2);
-    }
+        } catch (InterruptedException e) {
 
-    @Test
-    public void testElement3() throws Exception {
-        createNremoveMelementTest("/testElement3", 1000, 500);
+        }
+      }
+    };
+    offerThread.start();
+    offerThread.join();
+
+    takeThread.join();
+
+    Assert.assertTrue(takeResult[0] != null);
+    Assert.assertEquals(new String(takeResult[0], UTF_8), testString);
+  }
+
+  @Test
+  public void testTakeWait2() throws Exception {
+    String dir = "/testTakeWait2";
+    final String testString = "Hello World";
+    final int numClients = 1;
+    final ZooKeeper[] clients = new ZooKeeper[numClients];
+    final DistributedQueue[] queueHandles = new DistributedQueue[numClients];
+    for (int i = 0; i < clients.length; i++) {
+      clients[i] = createClient();
+      queueHandles[i] = new DistributedQueue(clients[i], dir, null);
     }
+    int numAttempts = 2;
+    for (int i = 0; i < numAttempts; i++) {
+      final byte[] takeResult[] = new byte[1][];
+      final String threadTestString = testString + i;
+      Thread takeThread = new Thread() {
+        public void run() {
+          try {
+            takeResult[0] = queueHandles[0].take();
+          } catch (KeeperException e) {
+
+          } catch (InterruptedException e) {
+
+          }
+        }
+      };
```

Comment:
Another conversion of lambda/multi-catch to the old style code that should go back to the way it was. There's more below. I'll stop calling them out, but they should be changed back everywhere.